### PR TITLE
[EOBS-546, 2360, 2378] Shift management bug fixes

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
Users who were not allocated directly to beds during shift change now correctly appear in the 'Nuring Staff on shift' field.

---

Before this pull request can be merged the following must be true.
- [ ] Unit tests pass (Travis integration).
- [ ] No code quality issues (Codacy integration).
- [ ] Approval from at least one developer and at least one tester.

If you are a BJSS contributor there are some additional conditions.
- [ ] All client module unit tests pass.
- [ ] The *Pull Request* field in the work tab of the JIRA issue is populated.
- [ ] The JIRA issue contains a description of the root cause and the solution in the comments.
- [ ] There are no bugs against the JIRA issues in the current sprint.

### CODE REVIEW

File: nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py [hunk 1]
```diff
@@ -1,67 +1,61 @@
...
-        users = self.allocation_pool._get_default_users(self.cr, self.uid)
-        self.assertEqual(users, self.USERS)
+        self.reallocation = self.reallocation_model.sudo(
+            self.shift_coordinator).create({})
+        actual_users_on_shift = self.reallocation.user_ids
+        self.assertEqual(self.expected_users_on_shift, actual_users_on_shift)
+
+    def test_returns_all_users_added_to_roll_call_shift_change_created(
+            self):
+        self.allocation_model.create({})
```
Comment (Nayira) [inline nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py]: Missing docstring

File: nh_clinical/tests/nh_clinical_staff_reallocation/test_fields_view_get.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from openerp.tests.common import TransactionCase
+
+
+# EOBS-549, EOBS-2378
+class TestFieldsViewGet(TransactionCase):
```
Comment (Nayira) [inline nh_clinical/tests/nh_clinical_staff_reallocation/test_fields_view_get.py]: missing docstring

File: nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py [hunk 3]
```diff
@@ -1,67 +1,61 @@
...
-        cls.allocation_pool._revert_method('_get_default_locations')
-        cls.users_pool._revert_method('search')
-        super(TestStaffReallocationDefaultUsers, cls).tearDownClass()
-
-    def test_calls_get_default_locations(self):
+from openerp.tests.common import TransactionCase
+
+
+# EOBS-549
+class TestStaffReallocationDefaultUsers(TransactionCase):
```
Comment (Nayira) [inline nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py]: missing docstring

File: nh_clinical/models/shift.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields
+
+
+class Shift(models.Model):
```
Comment (Nayira) [inline nh_clinical/models/shift.py]: missing docstring

File: nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py [hunk 5]
```diff
@@ -1,67 +1,61 @@
...
+
+    def setUp(self):
+        super(TestStaffReallocationDefaultUsers, self).setUp()
+        self.test_utils = self.env['nh.clinical.test_utils']
+        # Creates locations and users.
+        self.test_utils.admit_and_place_patient()
+        self.shift_coordinator = self.test_utils.create_shift_coordinator()
```
[the remark this answers is not in the document]
Comment (moderatemisbehaviour) [reply nh_clinical/tests/nh_clinical_staff_reallocation/test_default_users.py]: [Created.](https://jupiter.bjss.com/browse/EOBS-2384)

**Comment by Nayira** [comment]
Code review: LGTM :shipit:

**Pull request merged**